### PR TITLE
Devcontainer support

### PR DESCRIPTION
**Pull request by 4ydan**
## Devcontainers
I wanted ot share my devcontainers with you, since many use `code`. Maybe this is something others could benefit from, otherwise I can keep it for myself. Also this is kinda the byproduct of the base image for the CI containers later on. Related to #560 

### Benefits
- Attractive for (new) developers ready to go environment with a database connection and pgadmin
- Nice integration with vscode. Can clone/open pull request into the containers
- Already installed pre-commit for everyone developing inside the container
- [Github codespaces](https://github.com/features/codespaces)

### Downside
- Might be slower than developing on the host
- Building the image for the first time takes a bit of time
- Something additional that has to be "maintained" but removing it isn't really "breaking" anything

### Try it out
If you meet the prerequites for devcontainers you can try it out with this branch by opening vscode and cloning the pull request.
F1 Dev Containers: Clone Github Pull Request in Container Volume... and paste the URL of this pullrequest.

## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildserver is happy.

## Checklist

- [ ] I have installed and I am using [pre-commit hooks](../doc/contrib/README.md#Hooks)
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

File: .devcontainer/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+## Usage
+
+1. Ensure you have all [Devcontainer Prerequisites](#devcontainer-prerequisites)
+2. Install the [Remote Development](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack) extension pack for VS Code. This will be included if you install recommended workspace extensions upon opening this repository.
+3. Ensure Docker is running
+4. Make sure you are on the top level of the project. [`/PermaplanT`](https://github.com/ElektraInitiative/PermaplanT).
+5. [Open your workspace in the provided devcontainer](https://code.visualstudio.com/docs/remote/containers#_open-an-existing-workspace-in-a-container): Open this repository in VS Code and run **Remote-Containers: Reopen in Container...** from the Command Palette (<kbd>F1</kbd>).
+
+## Philosophy of the folder structure
```
Comment (absurd-turtle) [inline .devcontainer/README.md:24]:
```suggestion
## Philosophy of the directory structure
```

**Review by absurd-turtle — APPROVED** [review]
Looks pretty neat :)
I will maybe try the [devcontainer plugin](https://github.com/esensar/nvim-dev-container) for neovim.

**Comment by markus2330** [comment]
Thank you, seems like you already have a second user for it :wink:

**Comment by 4ydan** [comment]
There might be an issue with ./setup.sh not having the permissions.

**Pull request merged**